### DISCUSSION TITLE
Specify path for desktop entries

### DISCUSSION THREAD

**Discussion by Druidroyal** [davatorium/rofi#1835] [Q&A]
Hello,
I'm kinda new to the world of dektop customisation and i recently stared to use rofi for everything. But there's something i can't manage to do: specify to rofi where he has to find the desktop entries; I know it search automatically in the $PATH, but i want it to search in the folder i'll give him.

thanks in advence

**Comment by DaveDavenport** [comment]
Not sure what you mean with desktop entries.
but from the documentation:

run

Shows a list of executables in $PATH and can launch them (optional in a terminal). 

drun

Same as the run launches, but the list is created from the installed desktop files. 
The DRUN mode tries to follow the [XDG Desktop Entry Specification](https://freedesktop.org/wiki/Specifications/desktop-entry-spec/).
(this looks in the paths set by your system, via $XDG_DATA_PATHS)

**Reply by Druidroyal** [reply]
Hi,
Thanks for your answer, i already knew that. What i want is that rofi only displays executables that are in the folder that i will give.

**Reply by DaveDavenport** [reply]
PATH=myfolder rofi -show run?

**Reply by Druidroyal** [reply]
Something like that is possible?

**Reply by DaveDavenport** [reply]
It will scan that directory for binaries yes, and display them.

```bash
PATH=/home/druidroyal/bin/ /usr/bin/rofi -show run -disable-history
```

should work.

**Reply by Druidroyal** [reply]
I'll try that thanks

**Reply by Druidroyal** [reply]
It worked thanks

**Reply by ghost** [reply]
Hello, is it possible to do the same thing with the drun mode ?  Use a specifiec folder instead of /usr/share/applications/ or .local/share/applications etc.

Edit: after some research, i found that i have to edit the $XDG_DATA_DIRS, but it changes nothing, everytime i open rofi in drun mode, it searches in /usr/share/applications etc.

**Reply by DaveDavenport** [reply]
It uses `g_get_system_data_dirs`  See [here](https://developer-old.gnome.org/glib/stable/glib-Miscellaneous-Utility-Functions.html#g-get-system-data-dirs)

and uses `g_get_user_data_dir()` (see on same page as above) and this can be disabled in config.

**Reply by DaveDavenport** [reply]
```bash
XDG_DATA_DIRS=/dev/null rofi -show drun -disable-history -modes drun -drun-parse-user false
```
